pkg/collection/collection.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package collection provides high-level abstractions for collections of files
+package collection
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// MetadataType disambiguates different type of metadata structures between entry
+// implementations.
+type MetadataType uint8
+
+const (
+	CustomMetadataType MetadataType = iota
```

Comment:
This type is not used as Entry.Metadata() ignores its argument. I do not think that MetadataType is needed in this PR. If it is needed in the future, let's add it then.

Reply:
but what about the interface?

Reply:
We can change the interface, when we have different metadata types. For now, I do not see that we need more.